src/test/daml/Daml/Finance/Instrument/Generic/Test/CallableBond.daml

```diff
@@ -57,15 +57,17 @@ data TestParties = TestParties
     publicParty : Party
       -- ^ The public party. Every party can readAs the public party.
 
--- | Creates the claim for a callable bond and the corresponding elections
-bootstrapBond : Date -> Date -> Decimal -> Decimal -> Deliverable -> C
-bootstrapBond intermediateDate maturity couponAmount principalAmount ccy =
+-- | Creates the claim for a callable bond (with 2 coupons) and the corresponding elections
+bootstrapBond : Date -> Date -> Date -> Decimal -> Decimal -> Decimal -> Deliverable -> C
+bootstrapBond cpn1Date cpn2Date maturity couponAmount principalAmount callPrincipalAmount ccy =
   let
     coupon = scale (Const couponAmount) $ one ccy
     principal = scale (Const principalAmount) $ one ccy
-    called = ("CALLED",) $ give principal
-    notCalled = ("NOT CALLED",) $ when (at maturity) $ give principal
-    callableBond = when (at intermediateDate) $ and coupon $ give $ or called notCalled
+    callPrincipal = scale (Const callPrincipalAmount) $ one ccy
+    called = ("CALLED",) $ give callPrincipal
+    notCalledFinal = ("NOT CALLED",) $ when (at maturity) $ give principal
+    notCalled = ("NOT CALLED",) $ when (at cpn2Date) $ and coupon $ give $ or called notCalledFinal
```

Comment:
```suggestion
    notCalled = ("NOT CALLED",) $ when (at cpn2Date) $ and coupon $ called `or` notCalledFinal
```

Reply:
I think this gives the intended result, otherwise we would have a chain of two `give` nodes which gives the choice to the owner again

Reply:
I see, the `give` applies to everything afterwards, both payments and `or` nodes. In a callable bond, the `or` nodes "belong" to the issuer (the right to choose) and the payments belong to the investor. 

For consistency, it could be better to put a `give` before the very first node (the `and`) and then put a `give` in front of _every_ coupon (including the first one) and principal payment. Otherwise, we would have to special case the first coupon (no `give`) compared all other coupons (always `give`). Maybe not that relevant for this example with two coupons, but considering a dynamically created claim with _n_ coupons.

Reply:
From that point of view, a puttable bond would have been a simpler example to start with: no `give` nodes at all.

Reply:
> For consistency, it could be better to put a `give` before the very first node (the `and`) and then put a `give` in front of _every_ coupon (including the first one) and principal payment.

I have changed it in this way now.